pkg/relay/relay.go

```diff
@@ -216,6 +226,18 @@ func (rs *Relay) GetHeader(ctx context.Context, m *structs.MetricGroup, request
 		return nil, ErrInternal
 	}
 
+	if rs.config.Distributed { // TODO: run inside goroutine? If so, what context to use?
+		go func() {
```

Comment:
This is the endpoint with the highest performance requirements. The entire HTTP RTT must not exceed 900ms. So either we run it as a goroutine or set a default: skip the streaming. This is the best option. However, I'm open to talking with you about it.